src/server/database/migrations/20170406190200-addTags.js

```diff
@@ -0,0 +1,95 @@
+exports.up = async (r) => {
+  const mutations = [
+    r.table('Action'),
+    r.table('Project').replace((project) => {
+      return r.branch(
+        project('isArchived'),
+        project.merge({
+          tags: ['archived']
+        }).without('isArchived'),
+        project.merge({
+          tags: []
+        }).without('isArchived')
+      );
+    })
+  ];
+  const [actions] = await Promise.all(mutations);
+  const newProjects = actions.map((action, idx) => ({
+    id: action.id,
+    agendaId: action.agendaId,
+    content: action.content,
+    createdAt: action.createdAt,
+    isArchived: false,
+    sortOrder: idx, // meh, so they have to resort, oh well
+    status: 'active',
+    tags: ['private'],
+    teamId: action.teamMemberId.split('::')[1],
+    teamMemberId: action.teamMemberId,
+    updatedAt: action.updatedAt,
+  }));
+
+  await r.table('Project').insert(newProjects);
+
+  try {
+    await r.tableDrop('Action');
+  } catch (e) {
+    //
+  }
+
+  const indices = [
+    r.table('Project').indexCreate('tags', {multi: true}),
+  ];
+  try {
+    await Promise.all(indices);
+  } catch (e) {
+    // ignore
+  }
+
+  const waitIndices = [
+    r.table('Project').indexWait('tags'),
+  ];
+  await Promise.all(waitIndices);
+};
+
+exports.down = async (r) => {
+  const tables = [
+    r.tableCreate('Action')
+  ];
+  try {
+    await Promise.all(tables);
+  } catch (e) {
+    //
+  }
+  const projectsToConvert = await r.table('Project').getAll('private', {index: 'tags'});
```

Comment:
there's no way to get an idx in rethinkdb land that i've found out (that is less than nlogn).